src/shogun/machine/BaggingMachine.cpp

```diff
@@ -44,17 +47,53 @@ CBaggingMachine::~CBaggingMachine()
 
 CBinaryLabels* CBaggingMachine::apply_binary(CFeatures* data)
 {
-	SGVector<float64_t> combined_vector = apply_get_outputs(data);
+	SGMatrix<float64_t> output = apply_outputs_without_combination(data);
+
+	CMeanRule* mean_rule = new CMeanRule();
```

Comment:
This reminds me.
Please test your new unit tests using valgrind to check for memory errors
The dev readme has sections on that
`valgrind ctest -R unit-test_name***`